File: src/cpp/20_operacje_na_plikach/zad9.cpp
/*
Tytul: Przesun wszystkie pliki CSV do jednego folderu.
Tresc zadania: Otrzymujesz dwa napisy reprezentujace sciezki do folderow.
Przenies wszystkie pliki CSV z pierwszego folderu (oraz jego podfolderow) do
drugiego folderu. Dane wejsciowe: Dwa napisy reprezentujace sciezki do folderow.
Dane wyjsciowe: Brak.

*/
#include <cassert>
#include <experimental/filesystem>
#include <fstream>
#include <string>

namespace filesys = std::experimental::filesystem;

int rozmiarPliku(const std::string &sciezka) {
  return filesys::file_size(filesys::path(sciezka));
}

void usunPliki(const std::string &sciezka) {

  for (const auto plik : filesys::directory_iterator(sciezka)) {
    if (rozmiarPliku(plik.path()) > 10240)
      filesys::remove(filesys::path(plik.path()));
  }
}

void testUsunPliki() {

  // stworz folder testowy
  std::string sciezka = "test";
  filesys::create_directory(filesys::path(sciezka));

  // dodaj plik do folderu
  std::string sciezkaPliku = "test/plik.txt";
  std::ofstream plik(sciezkaPliku);
  for (int i = 0; i < 100000; i++)
    plik << "test\n";
  plik.close();

  // usun pliki z testowego folderu
  usunPliki(sciezka);

  // sprawdz czy folder test jest pusty
  assert(filesys::is_empty(filesys::path(sciezka)));

  // usun folder test
  filesys::remove_all(filesys::path(sciezka));
}

int main() {

  testUsunPliki();

  return 0;
}

File: src/python/20_operacje_na_plikach/zad9.py
"""
Tytul: Przesun wszystkie pliki CSV do jednego folderu.
Tresc zadania: Otrzymujesz dwa napisy reprezentujace sciezki do folderow. Przenies wszystkie pliki CSV z pierwszego folderu (oraz jego podfolderow) do drugiego folderu.
Dane wejsciowe: Dwa napisy reprezentujace sciezki do folderow.
Dane wyjsciowe: Brak.

"""
import pathlib

def usun_pliki(sciezka):
    """
    Funkcja usuwa pliki o rozmiarze wiekszym niz 10kB znajdujace sie
    w podanym folderze oraz jego podfolderach.
    """
    for sciezka_do_pliku in sciezka.iterdir():
        if sciezka_do_pliku.is_file():
            if sciezka_do_pliku.stat().st_size > 10240:
                sciezka_do_pliku.unlink()

def test_usun_pliki():

    # stworz folder testowy
    pathlib.Path("test").mkdir()

    # stworz pliki testowe
    plik = pathlib.Path("test/plik1.txt")
    plik.touch()
    plik.write_text("test\n" * 10000)

    assert plik.stat().st_size >= 10240

    # usun pliki
    usun_pliki(pathlib.Path("test"))

    # sprawdz czy plik zostal usuniety
    assert not plik.exists()

    # usun folder testowy
    import shutil

    shutil.rmtree("test")

if __name__ == "__main__":

    test_usun_pliki()

